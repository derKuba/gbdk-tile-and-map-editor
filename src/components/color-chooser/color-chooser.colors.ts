export enum Colors {
    WHITE = 'white',
    GREY = 'grey',
    GREY_DARK = 'grey-dark',
    BLACK = 'black',
}
