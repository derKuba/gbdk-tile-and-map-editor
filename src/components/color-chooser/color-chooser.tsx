import { Component, Host, h, Prop } from '@stencil/core';
import { Colors } from './color-chooser.colors';

import Tunnel from '../../provider/StateProvider';

@Component({
    tag: 'mvjp-color-chooser',
    shadow: true,
    styleUrl: 'color-chooser.css',
})
export class ColorChooser {
    private colors: string[] = ['white', 'grey', 'grey-dark', 'black'];

    /**
     *   choosen colorcode
     */
    @Prop() choosenColour: number;

    /**
     *   setter for colorcode
     */
    @Prop() updateColor: Function;

    private clickTile = color => {
        this.choosenColour = this.colors.indexOf(color);
        this.updateColor(this.choosenColour);
    };

    render() {
        return (
            <Host>
                <div class="color-chooser">
                    <mvjp-color-chooser-field
                        color={Colors.WHITE}
                        isActive={this.colors[this.choosenColour] === Colors.WHITE}
                        clickTile={this.clickTile}></mvjp-color-chooser-field>
                    <mvjp-color-chooser-field
                        color={Colors.GREY}
                        isActive={this.colors[this.choosenColour] === Colors.GREY}
                        clickTile={this.clickTile}></mvjp-color-chooser-field>
                    <mvjp-color-chooser-field
                        color={Colors.GREY_DARK}
                        isActive={this.colors[this.choosenColour] === Colors.GREY_DARK}
                        clickTile={this.clickTile}></mvjp-color-chooser-field>
                    <mvjp-color-chooser-field
                        color={Colors.BLACK}
                        isActive={this.colors[this.choosenColour] === Colors.BLACK}
                        clickTile={this.clickTile}></mvjp-color-chooser-field>
                </div>
            </Host>
        );
    }
}

Tunnel.injectProps(ColorChooser, ['choosenColour', 'updateColor']); // eslint-disable-line @stencil/ban-side-effects
