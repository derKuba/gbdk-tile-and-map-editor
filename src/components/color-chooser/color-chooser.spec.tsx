import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { ColorChooser } from './color-chooser';

describe('mvjp-color-chooser', () => {
    it('builds', () => {
        expect(new ColorChooser()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [ColorChooser],
            template: () => {
                return <mvjp-color-chooser choosenColour={0} updateColor={() => {}}></mvjp-color-chooser>;
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('click on clickhandler', async () => {
        const colorChooser = new ColorChooser();
        colorChooser.updateColor = jest.fn();

        colorChooser.clickTile('black');
        expect(colorChooser.choosenColour).toBe(3);
        expect(colorChooser.updateColor).toBeCalledWith(3);
    });
});
