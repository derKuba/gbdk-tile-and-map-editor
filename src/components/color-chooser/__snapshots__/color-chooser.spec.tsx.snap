// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-color-chooser matches snapshot 1`] = `
<mvjp-color-chooser>
  <mock:shadow-root>
    <div class="color-chooser">
      <mvjp-color-chooser-field color="white" isactive=""></mvjp-color-chooser-field>
      <mvjp-color-chooser-field color="grey"></mvjp-color-chooser-field>
      <mvjp-color-chooser-field color="grey-dark"></mvjp-color-chooser-field>
      <mvjp-color-chooser-field color="black"></mvjp-color-chooser-field>
    </div>
  </mock:shadow-root>
</mvjp-color-chooser>
`;
